newIDE/app/src/ProjectManager/index.js

```diff
@@ -235,1135 +572,912 @@ const ProjectManager = React.memo(
+                  ),
+                ];
+              }
+              return mapFor(
+                0,
+                project.getExternalEventsCount(),
+                i =>
+                  new LeafTreeViewItem(
+                    new ExternalEventsTreeViewItemContent(
+                      project.getExternalEventsAt(i),
+                      externalEventsTreeViewItemProps
+                    )
+                  )
+              );
+            },
+          },
+          {
+            isRoot: true,
+            content: new LabelTreeViewItemContent(
+              externalLayoutsRootFolderId,
+              i18n._(t`External layouts`),
+              {
+                icon: <Add />,
+                label: i18n._(t`Add an external layout`),
+                click: () => {
+                  // TODO Add after selected scene?
+                  const index = project.getExternalLayoutsCount() - 1;
+                  addExternalLayout(index, i18n);
+                },
+              }
+            ),
+            getChildren(i18n: I18nType): ?Array<TreeViewItem> {
+              if (project.getExternalLayoutsCount() === 0) {
+                return [
+                  new PlaceHolderTreeViewItem(
+                    externalLayoutEmptyPlaceholderId,
+                    i18n._(t`Start by adding new a external layout.`)
+                  ),
+                ];
+              }
+              return mapFor(
+                0,
+                project.getExternalLayoutsCount(),
+                i =>
+                  new LeafTreeViewItem(
+                    new ExternalLayoutTreeViewItemContent(
+                      project.getExternalLayoutAt(i),
+                      externalLayoutTreeViewItemProps
+                    )
+                  )
+              );
+            },
+          },
+        ].filter(Boolean);
       },
-      [project, onProjectItemModified]
+      [
+        addExternalEvents,
+        addExternalLayout,
+        addNewScene,
+        extensionTreeViewItemProps,
+        externalEventsTreeViewItemProps,
+        externalLayoutTreeViewItemProps,
+        onOpenGamesDashboardDialog,
+        onOpenPlatformSpecificAssets,
+        onOpenResources,
+        openProjectProperties,
+        openProjectVariables,
+        openSearchExtensionDialog,
+        project,
+        sceneTreeViewItemProps,
+      ]
     );
 
-    const moveDownEventsFunctionsExtension = React.useCallback(
-      (index: number) => {
-        if (index >= project.getEventsFunctionsExtensionsCount() - 1) return;
-
-        project.swapEventsFunctionsExtensions(index, index + 1);
-        onProjectItemModified();
-      },
-      [project, onProjectItemModified]
+    const canMoveSelectionTo = React.useCallback(
+      (destinationItem: TreeViewItem, where: 'before' | 'inside' | 'after') =>
+        selectedItems.every(item => {
+          return (
+            item.content.getRootId().length > 0 &&
```

Comment:
It feels a bit hacky. I don't understand why there is such a check when coming to moving it

Reply:
I added a comment, but maybe I should check the 4 root id: scene, external layout, external events and extension.

Reply:
It might be clear with the ids